File: demo/src/main/java/br/com/stickyindexsample/data_providers/SuggestionProvider.java
package br.com.stickyindexsample.data_providers;

import android.content.SearchRecentSuggestionsProvider;

public class SuggestionProvider extends SearchRecentSuggestionsProvider {

    public final static String AUTHORITY = "br.com.stickyindexsample.data_providers.SuggestionProvider";
    public final static int MODE = DATABASE_MODE_QUERIES;

    public SuggestionProvider() {
        setupSuggestions(AUTHORITY, MODE);
    }
}

File: demo/src/main/java/br/com/stickyindex/sample/data/SuggestionProvider.kt
package br.com.stickyindex.sample.data

import android.content.SearchRecentSuggestionsProvider

/**
 * Receive suggestions for queries in the database
 */
class SuggestionProvider : SearchRecentSuggestionsProvider() {
    init {
        setupSuggestions(SuggestionProvider::class.java.simpleName!!, DATABASE_MODE_QUERIES)
    }
}